portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -43,7 +44,16 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
+    String limit = request.getParameter("limit");
+    int limitNo;
+    if (limit == null) {
+      limitNo = results.countEntities(FetchOptions.Builder.withDefaults());
+    }
+    else {
+      limitNo = Integer.parseInt(limit);
```

Comment:
If you use a non-numeric 'limit' then parseInt will throw an exception, and you will probably wind up with a stack trace or other non-helpful error on the frontend.

You probably want to wrap this in a try-catch block and return a user-friendly error in the catch block (a'la "The limit value must be an integer, but is '12x'.").

Reply:
That makes sense, thanks @waxmittmann! I made the change